### ISSUE TITLE
User can escape the Message Dialog using keyboard or screen reader

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#1555]
### Provide a general summary of the issue here

When using keyboard or screen reader and navigating with tab, the message dialog modal is not a closed environment. When using keyboard or screen reader, the focus in an element with the role “dialog” should be contained in the element. Though a bit counterintuitive, this also includes the browser elements.

### 🤔 Expected Behavior?

When using keyboard or screen reader and navigating with tab, the focus stays inside the modal, and doesn’t “leak” outside of it to the browser elements or the web page below the modal.

### 😯 Current Behavior

The user can access the browser elements and the page below when a modal is open and using tab.

### 💁 Possible Solution

When in a modal the focus should be locked using the following method:
• Find the first and last focusable element programmatically
• When user has focus in the last element of the modal, if the user presses the “Tab”-key, move focus to the first focusable element
• When user has focus in the first element of the modal, if the user pressed “Shift + Tab” key combination, move focus to the last focusable element

### 🖥️ Steps to Reproduce

1. Open a page with the modal element (e.g. [Message dialog Alert](Open https://storybook.sanomalearning.design/?path=/story/overlay-message-dialog--alert)
2. Use the tab key (or shift+tab to reverse) to navigate the element (optionally start a screen reader)
3. When in the first or last focusable element of the modal, keep pressing tab (or shift-tab) to escape the modal

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

Windows 10

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning
```[tasklist]
### Tasks
- [ ] implement focus trap within message dialog
```
